node/network/collator-protocol/src/collator_side.rs

```diff
@@ -467,16 +467,13 @@ async fn connect_to_validators(
 	state: &mut State,
 	group: GroupValidators,
 )  {
-	match state.connection_handles.entry(group.group) {
-		Entry::Occupied(_) => {}
-		Entry::Vacant(vacant) => {
-			let (tx, rx) = mpsc::channel(0);
-			ctx.send_message(AllMessages::NetworkBridge(NetworkBridgeMessage::ConnectToValidators {
-				validator_ids: group.validators, peer_set: PeerSet::Collation, connected: tx
-			})).await;
-			vacant.insert(rx);
-		}
-	}
+	let (tx, rx) = mpsc::channel(0);
+	// Reconnect in all cases, as authority discovery cache might not have been fully populated
+	// last time:
+	ctx.send_message(AllMessages::NetworkBridge(NetworkBridgeMessage::ConnectToValidators {
+		validator_ids: group.validators, peer_set: PeerSet::Collation, connected: tx
+	})).await;
+	state.connection_handles.insert(group.group, rx);
```

Comment:
Generally looks good, I am not sure if we should explicitly call `.close()` on the potentially existing one and then consume the remaining items in the channel with `.try_next()`?

Reply:
it will be revoked on drop automatically, also we reference count, so submitting two identical requests is fine